### PR TITLE
ICU-21998 Update BRS instructions for updating online demos

### PR DESCRIPTION
**Pull request by sven-oly**
##### Checklist

- [X] Required: Issue filed: https://unicode-org.atlassian.net/browse/ICU-21998
- [X] Required: The PR title must be prefixed with a JIRA Issue number. 
- [X] Required: The PR description must include the link to the Jira Issue, for example by completing the URL in the first checklist item
- [X] Required: Each commit message must be prefixed with a JIRA Issue number. 
- [ ] Issue accepted (done by Technical Committee after discussion)
- [ ] Tests included, if applicable
- [ ] API docs and/or User Guide docs changed or added, if applicable

### CODE REVIEW

File: docs/processes/release/tasks/publish/index.md [hunk 1]
```diff
@@ -440,11 +440,27 @@ the new ICU version.
 
 ## Update online demos
 
-Update online demos/tools to the latest version:
+These are the online demos/tools that need to be updated to the latest version.
+
+* Be sure to verify that the deployed version is publicly available.
+
+Note that updating ICU4C demos onlin requires Gcloud access.
```
Comment (richgillam) [inline docs/processes/release/tasks/publish/index.md:447]: Typo: "online"

File: icu4c/source/i18n/dtitvfmt.cpp [hunk 2]
```diff
@@ -966,23 +966,26 @@ DateIntervalFormat::normalizeHourMetacharacters(const UnicodeString& skeleton) c
     
     UChar hourMetachar = u'\0';
     UChar dayPeriodChar = u'\0';
-    int32_t metacharStart = 0;
-    int32_t metacharCount = 0;
+    int32_t hourFieldStart = 0;
+    int32_t hourFieldLength = 0;
+    int32_t dayPeriodStart = 0;
+    int32_t dayPeriodLength = 0;
```
Comment (richgillam) [inline icu4c/source/i18n/dtitvfmt.cpp:972]: How'd all this stuff get in here?  This is from my PR a week or so ago that fixed a bug in DateIntervalFormat.  I had to cherry-pick it from the maint-70 branch over to main.  I'm guessing this all got included in your PR by mistake.

**Review by richgillam — CHANGES_REQUESTED** [review]
Your BRS instructions look fine to me (except for a typo), although I'm probably not the best person to review them.  But I suspect I'm included on this PR because it includes all the changes from my last PR, from a week ago, that fixed a bug in DateIntervalFormat.  It went onto the maint branch after 71 went out the door, so that it'd be there if we decided we needed a dot release.  I'm guessing you started this PR on the 71 code, not knowing that extra PR was there, or something like that...

File: docs/processes/release/tasks/publish/index.md [hunk 3]
```diff
@@ -440,11 +440,27 @@ the new ICU version.
...
+
+### ICU4C demos
+* [Run ICU4C demos](https://icu4c-demos.unicode.org/icu-bin/idnbrowser)
+
+* [Demo described here:](https://github.com/unicode-org/icu-demos/blob/main/README.md).
+
+* [Building and deploying from GCloud](https://github.com/unicode-org/icu-demos/blob/main/README.md).
+
+### UCU4J demos
```
Comment (srl295) [inline docs/processes/release/tasks/publish/index.md:456]:
```suggestion
### ICU4J demos
```

**Review by srl295 — COMMENTED** [review]
LGTM modulo extraneous changes

File: icu4c/source/i18n/dtitvfmt.cpp (hunk 2 above)
Comment (sven-oly) [reply icu4c/source/i18n/dtitvfmt.cpp:972]: Gack, my fault with Git. I'll remove these extra files from the PR.

Comment (richgillam) [reply icu4c/source/i18n/dtitvfmt.cpp:972]: I hate to say it, but my changes are still there.  I see my changes are showing as the first commit in this PR; maybe there's a way to rebase this on top of that so as to exclude that commit?

Comment (sven-oly) [reply icu4c/source/i18n/dtitvfmt.cpp:972]: Sorry to be slow fixing this PR. I'll try to fix this later today.

Comment (richgillam) [reply icu4c/source/i18n/dtitvfmt.cpp:972]: No biggie.  It's happened to all of us...

**Comment by sven-oly** [comment]
Closing this to try again without extraneous file changes.

**Pull request closed without merging**